types.go

```diff
@@ -74,13 +75,16 @@ type Level3Data struct {
 type AuthorizationRequest struct {
 	Amount                     Amount
 	CreditCard                 *CreditCard
-	BillingAddress             *BillingAddress
+	BillingAddress             *Address
 	Level3Data                 *Level3Data
 	ClientTransactionReference *string // Custom transaction reference metadata that will be associated with this request
 	Channel                    string  // for Psps that track the sales channel
 	Cryptogram                 string  // for Network Tokenization methods
 	ECI                        string  // E-Commerce Indicator (can be used for Network Tokenization as well)
 	MerchantOrderReference     string  // Similar to ClientTransactionReference but specifically if we want to store the shopping cart order id
+	ShippingAddress            *Address
+	ShopperIP                  *string
```

Comment:
Since this is a narrowly used piece of data, would it make sense to pass ShopperIP in Options? If it's needed for other processors down the line, we can pull it out into its own field at that point.

If we add fields to this object every time we need to pass one-off pieces of data, the object size is going to get out of hand.